lib/logstash/outputs/file.rb

```diff
@@ -198,9 +207,27 @@ def close_stale_files
   end
 
   private
-  def open(path)
-    return @files[path] if @files.include?(path) and not @files[path].nil?
+  def cached?(path)
+     @files.include?(path) && !@files[path].nil?
+  end
 
+  private
+  def deleted?(path)
+    !File.exist?(path)
+  end
+
+  private
+  def open(path)
+    if !deleted?(path) && cached?(path)
```

Comment:
this is a fair concert, however I added another "feature" here, if the file is deleted but ```@create_if_deleted``` is false, it appends the data to the failure file, like this there is no chance stuff is "lost". What do you think?